app.py

```diff
@@ -0,0 +1,383 @@
+#!/usr/bin/env python2.7
+
+"""
+Amity Room Allocation System
+Usage:
+  create_room <room_type> <room_name>
+  add_person <first_name> <second_name> <email> <FELLOW|STAFF> [wants_accommodation]
+  reallocate_person <email> <new_room_name>
+  load_people
+  print_allocations [-o=filename] 
+  print_unallocated [-o=filename]
+  print_room <room_name>
+  save_state [--db=sqlite_database] 
+  load_state <sqlite_database>
+Options:
+    -h, --help  Show this screen and exit
+"""
+
+import cmd
+import shutil
+import click
+from docopt import docopt, DocoptExit
+from src.amity import Amity
+from ui_additions import playSpinner
+
+term_size = shutil.get_terminal_size((80, 20))
+term_width = term_size[0]
+
+
+amity = Amity()
+
+
+def amity_docopt(func):
+    """
+    This decorator is used to simplify the try/except block and pass the result
+    of the docopt parsing to the called action.
+    """
+
+    def fn(self, arg):
+        try:
+            opt = docopt(fn.__doc__, arg)
+
+        except DocoptExit as e:
+            # The DocoptExit is thrown when the args do not match.
+            # We print a message to the user and the usage block.
+
+            print('Invalid Command!')
+            print(e)
+            return
+
+        except SystemExit:
+            # The SystemExit exception prints help message
+            # We do not need to do the print here.
+            return
+
+        return func(self, opt)
+
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+    return fn
+
+
+
+class AmityInteractive(cmd.Cmd):
+    click.secho( "Loading amity ...",fg ='cyan')
+    playSpinner()
+    click.secho('>' * term_width, fg='cyan', bold=True)
+    intro = '\n' \
+        + 'Welcome to Amity Room Allocation Application.\n' \
+        + 'Add people, create rooms, allocate and reallocate the rooms.\n' \
+        + '(type "help" for a list of commands.)\n'
+    click.secho(intro, fg="white", bold=True)
+    click.secho('>' * term_width, fg='yellow')
+    prompt = '(Amity) $ '
+
+    @amity_docopt
+    def do_create_room(self, args):
+        """
+        Creates a room(s) in amity
+        Usage:
+          create_room <room_type> <room_name> ...
+        """
+        try:
+            room_type = args['<room_type>']
+            valid = amity.validate_room_type(room_type)
```

Comment:
Shift most of your functionality inside class methods rather that doing it on the docopt file